src/buffer/mutable.rs

```diff
@@ -284,7 +211,7 @@ impl<T: NativeType> MutableBuffer<T> {
     #[inline]
     pub unsafe fn set_len(&mut self, len: usize) {
         assert!(len <= self.capacity());
```

Comment:
Why still use the assert here?
There is a similar `debug_assert` here https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#1264

and as it's marked `unsafe` already, I think this runtime check is not needed?